File: apps/admin_mobile/lib/application/use_cases/service_logs/complete_service_log_use_case.dart
import '../../../domain/entities/service_log.dart';
import '../../../domain/repositories/service_log_repository.dart';

class CompleteServiceLogUseCase {
  final ServiceLogRepository _repo;
  CompleteServiceLogUseCase(this._repo);

  Future<ServiceLog> call(int id) => _repo.complete(id);
}

File: apps/admin-v2/src/application/use-cases/service-logs/complete-service-log.use-case.ts
import type { ServiceLogRepository } from '@/domain/repositories/service-log.repository';

export class CompleteServiceLogUseCase {
  constructor(private repo: ServiceLogRepository) {}

  execute(id: string, leftOwing?: boolean) {
    return this.repo.complete(id, leftOwing);
  }
}
